frontend/src/lib/taxonomy.tsx

```diff
@@ -1078,10 +1078,16 @@ for (const [key, value] of Object.entries(CORE_FILTER_DEFINITIONS_BY_GROUP.event
     } else {
         CORE_FILTER_DEFINITIONS_BY_GROUP.person_properties[key] = value
     }
-}
-
-for (const key of Array.from(SESSION_PROPERTIES_ADAPTED_FROM_PERSON.keys())) {
-    CORE_FILTER_DEFINITIONS_BY_GROUP.sessions[key] = CORE_FILTER_DEFINITIONS_BY_GROUP.person_properties[key]
+    if (SESSION_INITIAL_PROPERTIES_ADAPTED_FROM_EVENTS.has(key)) {
+        CORE_FILTER_DEFINITIONS_BY_GROUP.sessions[`$initial_${key.replace(/^\$/, '')}`] = {
+            ...value,
+            label: `Initial ${value.label}`,
+            description:
+                'description' in value
+                    ? `${value.description} Data from the first event in this session.`
+                    : 'Data from the first event in this session.',
```

Comment:
Being very nitpicky, but just because I'm not sure how this works: Is it strictly the first event in this session, or the first value seen in this session (which might be from event number 97)?

Reply:
It's actually the first non-null value in general, BUT very soon we're going to be changing the client to send an empty string for all the campaign ids on every event, because otherwise you get weird combinations if you have multiple pageviews that set different props like:

first_visit?utm_source=google
second_visit?utm_medium=sms

gives you `{initial_utm_source: "google", initial_utm_medium: "sms"}` which obvious makes no sense

Reply:
So really if you are using posthog-js it *would* be based on the first event